### ISSUE TITLE
[@plone/components] DatePicker a11y/UX question

### ISSUE THREAD

**Issue by sneridagh** [plone/volto#7330]
Given the `@plone/components` DatePicker:

https://plone-components.readthedocs.io/latest/?path=/story/quanta-datepicker--default

which is based in the RAC DatePicker:

https://react-spectrum.adobe.com/react-aria/DatePicker.html

When clicking on the input segments of the widget (day/month/year) we were wondering if the dropdown calendar should be triggered. The dropdown calendar is not appearing until the button is clicked (or triggered with the keyboard).

We've been discussing this in the Salamina Sprint today, and the TL;DR is that the widget behavior is correct from the a11y point of view. 

@Wagner3UB could you please elaborate? Thanks!

/cc @tisto @ichim-david @JeffersonBledsoe @plone/volto-accessibility

**Comment by stevepiercy** [comment]
I absolutely hate it when clicking/tapping on the day, month, or year the date picker gets triggered. I want to edit that portion, either with arrow keys or entering digits. If I want a date picker, I'll click the calendar icon.

I see that changing years using the datepicker is painful for mouse users/keyboard-phobics. Old people will not click 12*age times to select their birthdate. I couldn't find any way to add a year select menu in the [Calendar](https://react-spectrum.adobe.com/react-aria/Calendar.html) component. I guess it's not supported? I've also seen datepickers that have something like this:

```text
<<  <  Today  >  >>
```

...where clicking double arrows goes to previous/next year and single arrows to previous/next month. However, I much prefer a select menu for year, especially when I need to enter date of birth, where it's a click-scroll-click instead of my age number of clicks to get to the correct year.

**Comment by JeffersonBledsoe** [comment]
I'd agree that we shouldn't open it on focus. From a WCAG conformance perspective, having the date picker open on focus and move the focus inside the picker would fail [SC 3.2.1](https://www.w3.org/WAI/WCAG21/Understanding/on-focus.html). You could not move the focus into the picker, but users don't really expect this and the experience becomes pretty poor as you can't trap focus within the picker, leading to other UX issues (e.g. user expects to know when they're using the date picker, closing the picker, easily returning back to the date input field, etc.)

**Comment by sneridagh** [comment]
FTR, I just realised that Google Forms widget also behaves as the RAC widget does. eg. it does not open the dropdown:

<img width="295" height="220" alt="Image" src="https://github.com/user-attachments/assets/a507f57c-1bc7-4846-b483-5305ca7445b6" />

**Comment by Wagner3UB** [comment]
@sneridagh Sorry for the delay. I was working on the DatetimeWidget for v19 and found this PR.

I agree with @JeffersonBledsoe. For a11y purposes, this behavior is unacceptable. You shouldn't change the component status without the user's consent, so the pop-up shouldn't open on focus. This is a general rule, not specific to this case, the golden rule is to always require an action, never perform it automatically.

However, it is reasonable to open the pop-up if one of those 3 fields is clicked, or if the content changes (the user types a number, not using the mouse). In that case, you can open the pop-up and warn the user using an `aria-live` region or a `role="status"` announcement, so screen readers are properly notified of the change.

In this context, the a11y concern that worries me is whether a validation message like "This is a required question" (similar to Google Forms) will be properly read by screen readers in our case.

Please take a look at this PR: https://github.com/plone/volto/pull/7980

The question raised by @stevepiercy is interesting, but I'm afraid it can't be fulfilled using our current libraries, that seems to be our only limitation for now.
